core/src/Cuda/Kokkos_Cuda_Instance.cpp

```diff
@@ -672,15 +685,13 @@ void Cuda::impl_initialize(InitializationSettings const &settings) {
   const auto &dev_info     = Impl::CudaInternalDevices::singleton();
 
   const struct cudaDeviceProp &cudaProp = dev_info.m_cudaProp[cuda_device_id];
+  Impl::CudaInternal::m_deviceProp      = cudaProp;
 
-  Impl::CudaInternal::m_cudaDev    = cuda_device_id;
-  Impl::CudaInternal::m_deviceProp = cudaProp;
-
-  Kokkos::Impl::cuda_device_synchronize(
-      "Kokkos::CudaInternal::initialize: Fence on space initialization");
+  KOKKOS_IMPL_CUDA_SAFE_CALL(cudaSetDevice(cuda_device_id));
+  KOKKOS_IMPL_CUDA_SAFE_CALL(cudaDeviceSynchronize());
```

Comment:
I guess we could remove but it wouldn't really be related to the intent of this pull request.

Reply:
can't use the wrapper here because it used to use the static m_cudaDev which isn't static anymore